server/chat-plugins/random-battles.ts

```diff
@@ -244,6 +244,377 @@ function CAP1v1Sets(species: string | Species) {
+	) {
+		buf += `<hr />`;
+		buf += Chat.getDataPokemonHTML(newSpecies, dex.gen, 'SSB');
+		let weighthit = 20;
+		if (newSpecies.weighthg >= 2000) {
+			weighthit = 120;
+		} else if (newSpecies.weighthg >= 1000) {
+			weighthit = 100;
+		} else if (newSpecies.weighthg >= 500) {
+			weighthit = 80;
+		} else if (newSpecies.weighthg >= 250) {
+			weighthit = 60;
+		} else if (newSpecies.weighthg >= 100) {
+			weighthit = 40;
+		}
+		const details: {[k: string]: string} = {
+			"Dex#": String(newSpecies.num),
+			Gen: String(newSpecies.gen) || 'CAP',
+			Height: `${newSpecies.heightm} m`,
+		};
+		details["Weight"] = `${newSpecies.weighthg / 10} kg <em>(${weighthit} BP)</em>`;
+		if (newSpecies.color && dex.gen >= 5) details["Dex Colour"] = newSpecies.color;
+		if (newSpecies.eggGroups && dex.gen >= 2) details["Egg Group(s)"] = newSpecies.eggGroups.join(", ");
+		const evos: string[] = [];
+		for (const evoName of newSpecies.evos) {
+			const evo = dex.getSpecies(evoName);
+			if (evo.gen <= dex.gen) {
+				const condition = evo.evoCondition ? ` ${evo.evoCondition}` : ``;
+				switch (evo.evoType) {
+				case 'levelExtra':
+					evos.push(`${evo.name} (level-up${condition})`);
+					break;
+				case 'levelFriendship':
+					evos.push(`${evo.name} (level-up with high Friendship${condition})`);
+					break;
+				case 'levelHold':
+					evos.push(`${evo.name} (level-up holding ${evo.evoItem}${condition})`);
+					break;
+				case 'useItem':
+					evos.push(`${evo.name} (${evo.evoItem})`);
+					break;
+				case 'levelMove':
+					evos.push(`${evo.name} (level-up with ${evo.evoMove}${condition})`);
+					break;
+				case 'other':
+					evos.push(`${evo.name} (${evo.evoCondition})`);
+					break;
+				case 'trade':
+					evos.push(`${evo.name} (trade${evo.evoItem ? ` holding ${evo.evoItem}` : condition})`);
+					break;
+				default:
+					evos.push(`${evo.name} (${evo.evoLevel}${condition})`);
+				}
+			}
+		}
+		if (!evos.length) {
+			details[`<font color="#686868">Does Not Evolve</font>`] = "";
+		} else {
+			details["Evolution"] = evos.join(", ");
+		}
+		buf += `<font size="1">${Object.keys(details).map(detail => {
+			if (details[detail] === '') return detail;
+			return `<font color="#686868">${detail}:</font> ${details[detail]}`;
+		}).join("&nbsp;|&ThickSpace;")}</font>`;
+	}
+	return buf;
+}
+
+function generateSSBInnateInfo(name: string, dex: ModdedDex, baseDex: ModdedDex) {
+	let buf = ``;
+	// Special casing for users whose usernames are already existing, i.e. Perish Song
+	let effect = dex.getEffect(name + 'user');
+	let longDesc = ``;
+	const baseAbility = Dex.deepClone(baseDex.getAbility('noability'));
+	// @ts-ignore hack to record the name of the innate abilities without using name
+	if (effect.exists && effect.innateName && (effect.desc || effect.shortDesc)) {
+		// @ts-ignore hack
+		baseAbility.name = effect.innateName;
+		if (effect.desc) baseAbility.desc = effect.desc;
```

Comment:
I don't fully understand this code but there are a lot of ts-ignores here? Are they really necessary?

Reply:
innatteName does not exist and is not proper attribute

Reply:
yeah they are if we use innateName, @KrisXV might have a less ts-ignore-y way tho

Reply:
You would just use casts, right? `baseAbility.name = effect.innateName as string;`?

Reply:
That wouldn't help because while it does declare a type, it's still not defined on `Condition`